server/handle_map_layer_zxy.go

```diff
@@ -183,10 +194,58 @@ func (req HandleMapLayerZXY) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	w.Header().Add("Content-Type", mvt.MimeType)
 	w.Header().Add("Content-Length", fmt.Sprintf("%d", len(pbyte)))
 	w.WriteHeader(http.StatusOK)
-	w.Write(pbyte)
+
+	_, err = w.Write(pbyte)
+	if err != nil {
+		log.Errorf("error writing tile z:%v, x:%v, y:%v - %v", req.z, req.x, req.y, err)
+	}
 
 	// check for tile size warnings
 	if len(pbyte) > MaxTileSize {
 		log.Infof("tile z:%v, x:%v, y:%v is rather large - %vKb", req.z, req.x, req.y, len(pbyte)/1024)
 	}
 }
+
+func (req *HandleMapLayerZXY) extractParameters(r *http.Request) (map[string]provider.QueryParameter, error) {
+	var params map[string]provider.QueryParameter
+	if req.Atlas.HasParams(req.mapName) {
+		params = make(map[string]provider.QueryParameter)
+		err := r.ParseForm()
+		if err != nil {
+			return nil, err
+		}
+
+		for _, param := range req.Atlas.GetParams(req.mapName) {
+			if r.Form.Has(param.Name) {
+				val, err := config.ParamTypeDecoders[param.Type](r.Form.Get(param.Name))
+				if err != nil {
+					return nil, err
+				}
+				params[param.Token] = provider.QueryParameter{
+					Token: param.Type,
+					SQL:   param.SQL,
+					Value: val,
+				}
+			} else if len(param.DefaultValue) > 0 {
+				val, err := config.ParamTypeDecoders[param.Type](param.DefaultValue)
+				if err != nil {
+					return nil, err
+				}
+				params[param.Token] = provider.QueryParameter{
+					Token: param.Type,
+					SQL:   "?",
```

Comment:
I don't think this is correct. Shouldn't we be using `param.SQL`?

Reply:
No, this branch handles the case when:
* No parameter was passed in a query
* There is a `default_value` configured

In this case we should just use the default value, so SQL becomes just `?`

Reply:
Well, no. We should use the default_value instead of the value that would be passed in the query param. The SQL shouldn't change. Otherwise, you'll break the SQL.

Reply:
Indeed, thanks for the catch!